activesupport/test/core_ext/duration_test.rb

```diff
@@ -158,6 +158,14 @@ def test_date_added_with_multiplied_duration
     assert_equal Date.civil(2017, 1, 3), Date.civil(2017, 1, 1) + 1.day * 2
   end
 
+  def test_date_added_with_divided_duration
+    assert_equal Date.civil(2017, 1, 3), Date.civil(2017, 1, 1) + 4.days / 2
```

Comment:
I would change this to
```ruby
assert_equal Date.civil(2017, 12, 31), Date.civil(2018, 1, 1) - (1.week / 7)
```
since `4.days / 2` works well with and without the changes.
See #33179

Reply:
Hmmm, not sure where that came from as I know my original was `1.week / 7` and was minused not added! I must have discarded the wrong chunk when composing this PR. Will fix!

I will also add some more targeted tests as well regarding `parts`.